js/kucoin.js

```diff
@@ -1135,9 +1152,14 @@ module.exports = class kucoin extends Exchange {
          * @description fetch the deposit address for a currency associated with this account
          * @param {string} code unified currency code
          * @param {object} params extra parameters specific to the kucoin api endpoint
+         * @param {string|undefined} params.network the blockchain network name
          * @returns {object} an [address structure]{@link https://docs.ccxt.com/en/latest/manual.html#address-structure}
          */
         await this.loadMarkets ();
+        const version = this.options['versions']['private']['GET']['deposit-addresses'];
```

Comment:
@Dan-krm hum, this is tricky because v2 is the default value.

This is not great but why not change it before the request and restore it immediately after?

```Javascript
const oldValue = this.options['versions']['private']['GET']['deposit-addresses'];
this.options['versions']['private']['GET']['deposit-addresses'] = 'v2'
// request here
this.options['versions']['private']['GET']['deposit-addresses'] = oldValue
```

Reply:
Yeah that works

Reply:
@Dan-krm I'm having second thoughts now, because this is not really "thread safe" 🤔

Reply:
I applied the changes to both methods, I'm not sure if that will make it more thread safe or not, I could revert the changes to throw errors and require the user to adjust the version depending on the method used, or maybe reorganizing the api dictionary would be best.

Let me know what you think